low-level/framework/utils/sysfs_interface.py

```diff
@@ -41,8 +44,59 @@ def __init__(self):
         self.sas_phy_dir_list = []
         self.phy_dir_dict = {}
         self.sas_phy_dir_path = {}
+        self.host_base_dir = None
+        self.host_type = OSUtils.get_host_type()
+
+
+    def get_host_list(self):
+        """
+        Returns host list from host base directory.
+        /sys/class/fc_host or /sys/class/scsi_host
+
+        Example: ['host1', 'host2']
+        """
+
+        hosts = []
+
+        # Refresh host_type
+        self.host_type = OSUtils.get_host_type()
+
+        if self.host_type:
+            self.host_base_dir = os.path.join(self.sysfs, self.host_type)
+            #self.host_base_dir = os.path.join('/tmp/class', self.host_type)
+            try:
+                hosts = os.listdir(self.host_base_dir)
+            except FileNotFoundError:
+                # HBA card is not detected
+                self.host_type = None
+
+        hosts = [host for host in hosts if 'host' in host]
+
+        return hosts
+
+
+    def get_host_data(self, host, attr):
+        """
+        Returns value of the host attribute like port_name, state,
+        unique_id, etc..
+        """
+
+        if not self.host_type:
+            return
+
+        host_dir_path = os.path.join(self.host_base_dir, host)
+
+        try:
+            with open(os.path.join(host_dir_path, attr)) as cFile:
+                data = cFile.read().strip()
+        except FileNotFoundError as err:
+            raise HBAError(err.errno, "Failed to get '%s' due to '%s, %s'",
+                attr, err.strerror, err.filename)
+
+        return data
+
 
-    def initialize(self):
+    def initialize_sas_phy(self):
         """Method for initialization.
```

Comment:
have you change other places function name where this function is called in existing code ?

Reply:
Yes, only one place required this change. Please check this PR for low-level/sensors/impl/generic/node_sas_port.py

Reply:
Done.